src/AspNet.Security.OAuth.LinkedIn/LinkedInAuthenticationHandler.cs

```diff
@@ -24,7 +24,12 @@ public LinkedInAuthenticationHandler([NotNull] HttpClient client)
 
         protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity,
             [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens) {
-            var request = new HttpRequestMessage(HttpMethod.Get, Options.UserInformationEndpoint);
+            var endpoint = string.Empty;
+            if (Options.Fields.Count > 0) {
+                endpoint = $"{Options.UserInformationEndpoint}/~:({string.Join(",", Options.Fields)})";
```

Comment:
I see what you are saying. Yeah, LI strange url syntax throws a wrench into it. I will wait for @jerriep to weigh in before trying to do anything with this.